tests/neg/missing-implicit.scala

```diff
@@ -0,0 +1,16 @@
+import Predef.{byte2Byte => _}
+import math.Numeric
+
+def consume[T: Numeric](xs: List[T], limit: T): List[T] = xs match
+  case x :: xs1 if limit > 0 => consume(xs1, limit - x) // error // error
+  case _ => xs
```

Comment:
This test case seems to already exercise conditional extension methods, which I believe is the most important thing to address, but could we also add a more complex test case like the following?

~~~ scala
trait Zip[F[_]]
trait Traverse[F[_]] {
  def [A, B, G[_] : Zip](fa: F[A]) traverse(f: A => G[B]): G[F[B]]
}

object instances {
  given zipOption: Zip[Option] = ???
  given traverseList: Traverse[List] = ???
}

List(1, 2, 3).traverse(x => Option(x)) // error
~~~

Reply:
I added the test.